__tests__/app.test.js

```diff
@@ -130,7 +130,29 @@ describe("GET /api/articles", () => {
     return request(app)
       .get("/api/articles")
       .then(({ body }) => {
-        expect(body).toBeSortedBy("created_at", { descending: true });
+        const { articles } = body;
+        expect(articles).toBeSortedBy("created_at", {
+          descending: true,
+        });
+      });
+  });
+  test("should return a list of articles filtered by topic query", () => {
+    return request(app)
+      .get("/api/articles?topic=mitch")
+      .then(({ body }) => {
+        const { articles } = body;
+        expect(articles).toHaveLength(12);
+        articles.forEach((article) => {
+          expect(article.topic).toBe("mitch");
+        });
+      });
+  });
+  test("should return a 404 error if topic does not exist", () => {
```

Comment:
There is a more nuanced test needed here and it is quite easy to miss. What about a situation where there is a topic but there are no articles for it? How would handle this?